File: _designs_001/Implement Stack using Queues/Solution.java
public class Solution {
    public List<Integer> testMyStack(StreamTokenizer tokenizer, List<Double> metrics) throws IOException {
        MyStack myStack = new MyStack();
        List<Integer> result = new ArrayList<Integer>();

        int opsCnt = Serializer.deserializeInt(tokenizer);
        for (int i = 0; i < opsCnt; i++) {
            int n;
            double startTime;
            switch (Serializer.deserializeInt(tokenizer)) {
            case 0:
                n = Serializer.deserializeInt(tokenizer);
                startTime = System.nanoTime();
                myStack.push(n);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 1:
                startTime = System.nanoTime();
                myStack.pop();
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 2:
                startTime = System.nanoTime();
                result.add(myStack.top());
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            case 3:
                startTime = System.nanoTime();
                result.add(myStack.empty() ? 1 : 0);
                metrics.set(0, metrics.get(0) + (double) (System.nanoTime() - startTime) / 1000000L);
                break;
            }
        }
        return result;
    }
}

File: _designs_001/Implement Stack using Queues/Solution.rb
class Solution
  def testMyStack(inp, metrics)
    metrics[0] = 0
    stk, r = MyStack.new, []
    Serializer.deserializeInt(inp).times do
      case Serializer.deserializeInt(inp)
      when 0
        n = Serializer.deserializeInt(inp)
        start_time = Time.now
        stk.push(n)
        metrics[0] += (Time.now - start_time) * 1000
      when 1
        start_time = Time.now
        stk.pop
        metrics[0] += (Time.now - start_time) * 1000
      when 2
        start_time = Time.now
        r << stk.top
        metrics[0] += (Time.now - start_time) * 1000
      when 3
        start_time = Time.now
        r << (stk.empty == true ? 1 : 0)
        metrics[0] += (Time.now - start_time) * 1000
      end
    end
    r
  end
end
